### ISSUE TITLE
Compiling Layout files breaks Intellisense

### ISSUE THREAD

**Issue by jeremyalan** [RazorGenerator/RazorGenerator#47]
I'm not sure if this is a bug, or if I'm just missing a step in the setup process...

I'm using this project to share common views between several ASP.NET MVC applications (using ASP.NET MVC 4).  The most common scenario I would like to take advantage of is having a set of layouts defined in a shared library, and be able to reference them from each of our applications.

For example, I might have a line in one of my application views, that looks like:
```
@{
   Layout = "~/Views/Layouts/_SideBarMenuLayout.cshtml";
}
```

The file being referenced here exists in the library, not in the application directory.  This works great when I run the application, but I don't get any Intellisense in the .cshtml file, and I have red squigglies around all my `@section` declarations, because the Layout file cannot be deteremined at design-time.

Is this something that can be achieved?

**Comment by kfisher723** [comment]
I'm running into an identical issue with my project in VS2015.  In particular, the error states that "The type or namespace name 'System' could not be found (are you missing a using directive or an assembly reference?)".  This error is always indicated on line 1 of the .cshtml file, but doesn't prevent compilation except on complete rebuild.

**Comment by pranavkm** [comment]
Do you see this issue when you're not using RazorGenerator (a brand new Mvc application?) If so it might just be the tooling being broken.

**Comment by jeremyalan** [comment]
@pranavkm No, it works fine for an MVC project that is not split across multiple assemblies.  It appears the issue is caused because the Layout path is actually matched by the EmbeddedResourceVirtualPathProvider, which is not supported by Intellisense.

e.g.
```cs
@{
   // This path is fulfilled by an embedded resource, the actual file does not exist.
   Layout = "~/Views/Layouts/_SidebarMenuLayout.cshtml";
}
```

**Comment by gregsdennis** [comment]
I also am experiencing this.  Commenting to bump the issue as it hasn't been visited in a couple years. 😕
